client/src/key.rs

```diff
@@ -47,6 +47,14 @@ impl Key {
         Ok(Self::new(crypto::generate_public_key(private_key)?))
     }
 
+    pub fn generate_public_key_for_secret_exchange(
```

Comment:
Those new methods needs tests, similar to the previous ones to ensure we have the same behaviour in both native & openssl crypto.

Reply:
Done.